numpy/lib/function_base.py

```diff
@@ -1482,42 +1482,57 @@ def angle(z, deg=False):
     return a
 
 
-def _unwrap_dispatcher(p, discont=None, axis=None):
+def _unwrap_dispatcher(p, discont=None, axis=None, *, interval_size=2*pi, min_val=None, max_val=None):
     return (p,)
 
 
 @array_function_dispatch(_unwrap_dispatcher)
-def unwrap(p, discont=pi, axis=-1):
+def unwrap(p, discont=None, axis=-1, *, interval_size=2*pi, min_val=None, max_val=None):
     """
-    Unwrap by changing deltas between values to 2*pi complement.
-
-    Unwrap radian phase `p` by changing absolute jumps greater than
-    `discont` to their 2*pi complement along the given axis.
-
+    Unwrap by changing deltas between values to complement.
+    
+    For the default case where `interval_size= 2*pi`, `discont=pi`,
+    It unwraps radian phase `p` by changing absolute jumps greater 
+    than `discont` to their 2*pi complement along the given axis.
+    
+    In general it unwrapps a signal `p` by changing absolute jumps 
+    greater than `discont` to their `interval_size` complementary values.
+    
     Parameters
     ----------
     p : array_like
         Input array.
     discont : float, optional
-        Maximum discontinuity between values, default is ``pi``.
+        Maximum discontinuity between values, default is ``interval_size/2``.
     axis : int, optional
         Axis along which unwrap will operate, default is the last axis.
-
+    interval_size: float, optional
+        Size of the range over which the input wraps. By default, it is 2 pi.
+        If ``min_val`` and ``max_val`` are given, ``interval_size`` is ignored
+        and the interval size is ``max_val - min_val``. 
+    min_val, max_val: float, optional
+        Boundaries of the interval over which the input array is expected to 
+        wrap. By default, they are ``None`` and the interval is considered as 
+        ``[-interval_size, interval_size]``. In case the first value of the 
+        phase input array, ``p[0]``, is outside of the interval 
+        ``[min_val, max_val]`` it will be corrected by an integral multiple of 
+        the interval size such that it will be within the 
+        boundaries.
+        Both boundaries require each other. If only one boundary is 
+        provided without the other, it will be ignored. 
```

Comment:
Apologies for the wild goose chase - I now agree with your argument, and think we should just remove these.